src/contributing/release_process.md

```diff
@@ -0,0 +1,164 @@
+# The Parsec Release Process
+
+This page describes the process by which the Parsec project is released, and sets expectations
+around the schedule of releases so that consumers know when to expect them.
+
+## Release Schedule
+
+The Parsec project aims to make major releases twice a year: in early March and early September.
+Precise dates are agreed by the maintainers and the project community on each cycle. A fixed release
+cadence helps with the process of packaging Parsec releases into Operating System distributions and
+other downstream consumers.
+
+## Responsibility for Releases
+
+It is the collective responsibility of the [Parsec project
+maintainers](https://github.com/parallaxsecond/parsec/blob/main/MAINTAINERS.toml) to ensure that
+Parsec releases are made according to the schedule and process outlined on this page. However, all
+project contributors have a role to play in ensuring that the release cycles are timely and
+successful. All contributors are encouraged to familiarise themselves with the release process, and
+all are expected to abide by its associated rules.
+
+## Release Roadmap
+
+A public roadmap for the Parsec project is under construction. In the meantime, the [Project
+dashboard](https://github.com/orgs/parallaxsecond/projects/1) can be used to track the items that
+are being prepared for the next release. Use the
+[milestones](https://github.com/parallaxsecond/parsec/milestones) to filter the dashboard view in
+order to see the issue tickets that have been targeted for each release.
+
+Please be aware that the content of a release cannot be guaranteed. In order to ensure a fixed
+release schedule, it is sometimes necessary to modify the scope of each release. The project
+dashboard always provides the best available indication of the current plan.
+
+## Released Components
```

Comment:
If we wanted to make it official, I think it would also be nice to indicate that a quickstart archive is also released as a GitHub Asset (see [here](https://github.com/parallaxsecond/parsec/releases/tag/0.7.0) at the bottom for example) with each version. This is to work with the [quickstart](https://parallaxsecond.github.io/parsec-book/getting_started/linux_x86.html) guide.